code/modules/surgery/limb_reattach.dm

```diff
@@ -51,79 +123,93 @@
+	"<span class='alert'>Your hand slips, damaging [target]'s [E.amputation_point]!</span>")
+	target.apply_damage(10, BRUTE, null, sharp=1)
+	return 0
 
 /datum/surgery_step/limb/mechanize
+	name = "attach robotic limb"
 	allowed_tools = list(/obj/item/robot_parts = 100)
 
-	min_duration = 80
 	max_duration = 100
 
-	can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		if(..())
-			var/obj/item/robot_parts/p = tool
-			if (p.part)
-				if (!(target_zone in p.part))
-					return 0
-			return isnull(target.get_organ(target_zone))
-
-	begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+/datum/surgery_step/limb/mechanize/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	if(..())
+		var/obj/item/robot_parts/p = tool
+		if (p.part)
+			if (!(target_zone in p.part))
+				return 0
+		return isnull(target.get_organ(target_zone))
+
+/datum/surgery_step/limb/mechanize/begin_step(mob/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
 		user.visible_message("[user] starts attaching \the [tool] to [target].", \
 		"You start attaching \the [tool] to [target].")
 
-	end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		var/obj/item/robot_parts/L = tool
-		user.visible_message("<span class='notice'>[user] has attached \the [tool] to [target].</span>",	\
-		"<span class='notice'>You have attached \the [tool] to [target].</span>")
-
-		if(L.part)
-			for(var/part_name in L.part)
-				if(!isnull(target.get_organ(part_name)))
-					continue
-				var/list/organ_data = target.species.has_limbs["[part_name]"]
-				if(!organ_data)
-					continue
-				var/new_limb_type = organ_data["path"]
-				var/obj/item/organ/external/new_limb = new new_limb_type(target)
-				new_limb.robotize(L.model_info)
-				if(L.sabotaged)
-					new_limb.sabotaged = 1
-
-		target.update_body()
-		target.updatehealth()
-		target.UpdateDamageIcon()
-
-		qdel(tool)
-
-	fail_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-		user.visible_message("<span class='alert'>[user]'s hand slips, damaging [target]'s flesh!</span>", \
-		"<span class='alert'>Your hand slips, damaging [target]'s flesh!</span>")
-		target.apply_damage(10, BRUTE, null, sharp=1)
+/datum/surgery_step/limb/mechanize/end_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+	var/obj/item/robot_parts/L = tool
+	user.visible_message("<span class='notice'>[user] has attached \the [tool] to [target].</span>",	\
+	"<span class='notice'>You have attached \the [tool] to [target].</span>")
+
+	if(L.part)
+		for(var/part_name in L.part)
+			if(!isnull(target.get_organ(part_name)))
+				continue
+			var/list/organ_data = target.species.has_limbs["[part_name]"]
+			if(!organ_data)
+				continue
+			var/obj/item/organ/external/stump = target.organs_by_name["limb stump"]
```

Comment:
I still think you're going to have issues here if the person you're operating on has more than one stump. organs_by_name will only hold one entry under "limb stump", so when someone ends up with another stump, the first one is overwritten. Really you need the stump name to be added to in it's new() identifying which target zone it's in so they're added to organs_by_name individually and then have something like 
'var/obj/item/organ/external/stump = target.organs_by_name["[target_zone] limb stump"]'